jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -4,6 +4,25 @@
 from ..stpipe import Step, cmdline
 from .. import datamodels
 from . import outlier_detection
+from . import outlier_detection_scaled
+from . import outlier_detection_ifu
+from . import outlier_detection_spec
+from ..resample import resample, resample_spec
+
+# Categorize all supported versions of outlier_detection
+outlier_registry = {'imaging' : outlier_detection.OutlierDetection,
+                    'scaled' : outlier_detection_scaled.OutlierDetectionScaled,
+                    'ifu' : outlier_detection_ifu.OutlierDetectionIFU,
+                    'slitspec' : outlier_detection_spec.OutlierDetectionSpec
+                    }
+
+# Categorize all supported modes
+IMAGE_MODES = ['NRC_IMAGE', 'MIR_IMAGE', 'NRS_IMAGE', 'NIS_IMAGE', 'FGS_IMAGE']
```

Comment:
I don't think NRS_IMAGE exposures will ever make it to level-3 processing (just used as singletons for target-acq and confirmation images), but it can't hurt to include it in the list, just in case.

Reply:
This allows other packages to get the full categorization of exptypes, while also covering all possible options for currently implemented modes.